### PR TITLE
Updates terraform devcontainer dockerfile & configuration

### PR DESCRIPTION
**Pull request by wbreza**
Moves devcontainer files to `templates/common` to follow conventions

- [x] Includes use of terraform feature in devcontainer.json
- [x] Feature allows us to deprecate custom docker file

### CODE REVIEW

File: templates/todo/projects/python-mongo/.repo/terraform/azure.yaml [hunk 1]
```diff
@@ -15,5 +15,3 @@ services:
     project: ../../api/python
     language: py
     host: appservice
-infra:
```
Comment (ellismg) [inline templates/todo/projects/python-mongo/.repo/terraform/azure.yaml:18]: Did you intend this - it seems wrong?

Comment (ellismg) [reply templates/todo/projects/python-mongo/.repo/terraform/azure.yaml:18]: Oh - I see - It's also higher in this file :-)

**Review by ellismg — APPROVED** [review]
LGTM - (Even if the diff was trying to trick me!)

File: templates/todo/projects/python-mongo/.repo/terraform/azure.yaml (hunk 1 above)
Comment (wbreza) [reply templates/todo/projects/python-mongo/.repo/terraform/azure.yaml:18]: Somehow it got in there twice.  This remove the bottom one.

**Comment by jongio** [comment]
Do we also need for node?

**Pull request merged**